source/usb/usbd_core.c

```diff
@@ -879,6 +894,9 @@ void USBD_EndPoint0(U32 event)
             case REQUEST_CLASS:
                 switch (USBD_SetupPacket.bmRequestType.Recipient) {
                     case REQUEST_TO_DEVICE:
+                        if (USBD_EndPoint0_Setup_HID_ReqToIF()) {
```

Comment:
Instead of creating new hid interface this approach is using device control transfer to send/receive data from the device.